EvernoteOAuth.lua

```diff
@@ -1,4 +1,4 @@
-
+local logger = require('logger')
```

Comment:
logger is a KOReader thing, pinging @NiLuJe

Reply:
It is, but once koreader is shipped `EvernoteOAuth.lua` is moved to `koreader/plugins/evernote.koplugin`, so `logger` is available there.

Reply:
Of course, but that wouldn't be true for someone else. Changes here should generally be considered similar to contributing a fix to lua-rapidjson or something. Also see for example https://github.com/koreader/koreader/pull/4580 or more historically https://github.com/koreader/kindlepdfviewer/pull/825 (and of course https://github.com/koreader/android-luajit-launcher).

Reply:
Which doesn't necessarily mean we can't use logger, but it means being slightly less direct about it.

Like over here, you could pass a reference to logger.dbg (and possibly set some debug output switch to true?)
https://github.com/koreader/koreader/blob/1abaa39c27a4a49d1209a0602ff13ff802b07de3/plugins/evernote.koplugin/main.lua#L404-L408

Over here you could add something like `logger = print` as a default.
https://github.com/koreader/evernote-sdk-lua/blob/c9ad262784ea495659bd63ea58bfc561c654e5ec/EvernoteOAuth.lua#L98-L135

Reply:
Not sure if it's alright, I used `require` with `pcall`, if `require('logger')` fails, I reassign `logger.dbg` as `print`

Reply:
Functionally it would behave mostly as desired, but I think it makes the code harder to follow. What I suggested is a more flexible, typical way of dealing with Lua objects.

See https://www.lua.org/pil/16.2.html and how all the widgets work (like in https://github.com/koreader/koreader/blob/f8af0a9e0e00b15a978ae3849438c17264b28110/frontend/ui/widget/widget.lua#L19-L50).

The instantiation would then look like something like this:
```lua
    local oauth = EvernoteOAuth:new{
        domain = self.evernote_domain,
        username = username,
        password = password,
        logger = logger.dbg,
    }
```
That leaves the user open to plug in any logging function they like.

Reply:
Should be fixed now.